docs/new-architecture-backward-turbomodule.md

```diff
@@ -0,0 +1,699 @@
+
+### Test the Native Module
+
+Now we have all our code set up, it's time to test that everything work.
+
+As first thing, let's navigate to the `OldArchitectureApp` we created [here](#create-a-testing-environment). Then, let's run the following command, to add the library as a new dependency:
+
+```sh
+yarn add ../example-library
+```
+
+Now, on iOS, let's install the pod:
+
+```sh
+cd ios && pod install && cd ..
+```
+
+If everything goes right, we should see something like this:
+
+```sh
+Auto-linking React Native module for target `OldArchitectureApp`: example-library
+Analyzing dependencies
+Downloading dependencies
+Installing example-library (0.1.0)
+Generating Pods project
+Integrating client project
+Pod installation complete! There are 55 dependencies from the Podfile and 45 total pods installed.
+```
+
+Let's now open the `App.js` file and update to code to use our new library. Let's replace the code with the following:
+
+```js title="App.js"
+/**
+ * @format
+ * @flow strict-local
+ */
+
+import React from 'react';
+import { useEffect, useState } from 'react';
+import type { Node } from 'react';
+import {
+  SafeAreaView,
+  StatusBar,
+  Text,
+  View,
+  Button
+} from 'react-native';
+
+import type { AuthorizationStatus } from 'example-library/src/index';
+import AuthorizationManager from 'example-library/src/index';
+
+const App: () => Node = () => {
+  const [authorization, setAuthorization] =
+    useState<AuthorizationStatus>('NotDetermined');
+
+  useEffect(() => {
+    async function setInitialState() {
+      const newState =
+        await AuthorizationManager.getCurrentStatus();
+      setAuthorization(newState);
+    }
+    setInitialState();
+  }, []);
+
+  return (
+    <SafeAreaView>
+      <StatusBar barStyle="dark-content" />
+
+      <Button
+        title="Request Authorization"
+        onPress={() => {
+          AuthorizationManager.askAuthorization();
+        }}
+      />
+
+      <Text style={{ marginLeft: 20 }}>
+        Current authorization is: {authorization}
+      </Text>
+    </SafeAreaView>
+  );
+};
+
+export default App;
+```
+
+We can now go back to the terminal and start Metro:
+
+```sh
+npx react-native start
+```
+
+and then run the app:
+
+```sh
+npx react-native run-ios
+```
+
+After the app launches, we should see the following state in the simulator:
+
+<figure>
+  <img src="docs/assets/Turbomodules/OldArchApp.png" />
+  <figcaption>The simulator running the app</figcaption>
+</figure>
+
+But, if we tap on the button, nothing happens. That's because the `Core Location` system requires some keys in the plist. Let's do the following:
+
+1. Open the `ios/OldArchitectureApp.xcworkspace`.
+1. Select the project in the project navigator.
+1. Select the info tab.
+1. Scroll down and add the following two keys:
+   - `NSLocationWhenInUseUsageDescription`
+   - `NSLocationAlwaysAndWhenInUseUsageDescription`
+1. Add any description you want for each of these.
```

Comment:
This is really specific to `Core Location`. As this page is already long, shouldn't we try to simplify the TM to do something easier (say write on the log)?

I know it won't be as realistic as this example, but it would be easier to consume as a guide.

Reply:
Yeah. I thought about this. I wanted to find an example that makes sense for both iOS and Android. 
Unfortunately, `RCTLog` does not work with Hermes and Turbomodules and it limits a little bit the possibilities to showcase what we can do with a TM.
What about a calculator, instead? So we can show also return types an parameters? 🤔